src/main/java/org/logstash/beats/V2Batch.java

```diff
@@ -80,25 +90,82 @@ public int getHighestSequence(){
 
     /**
      * Adds a message to the batch, which will be constructed into an actual {@link Message} lazily.
-      * @param sequenceNumber sequence number of the message within the batch
+     * @param sequenceNumber sequence number of the message within the batch
      * @param buffer A ByteBuf pointing to serialized JSon
      * @param size size of the serialized Json
      */
     void addMessage(int sequenceNumber, ByteBuf buffer, int size) {
         written++;
-        if (internalBuffer.writableBytes() < size + (2 * SIZE_OF_INT)){
-            internalBuffer.capacity(internalBuffer.capacity() + size + (2 * SIZE_OF_INT));
+        if (internalBuffer.writableBytes() < size + (2 * SIZE_OF_INT)) {
+            int requiredSize = internalBuffer.capacity() + size + (2 * SIZE_OF_INT);
+            eventuallyLogIdealMaxOrder(requiredSize, logger);
+
+            internalBuffer.capacity(requiredSize);
         }
         internalBuffer.writeInt(sequenceNumber);
         internalBuffer.writeInt(size);
         buffer.readBytes(internalBuffer, size);
-        if (sequenceNumber > highestSequence){
+        if (sequenceNumber > highestSequence) {
             highestSequence = sequenceNumber;
         }
     }
 
+    // package-private for testability reasons
+    void eventuallyLogIdealMaxOrder(int requiredSize, Logger logger) {
+        int idealMaxOrder = idealMaxOrder(requiredSize);
+        if (idealMaxOrder <= PooledByteBufAllocator.defaultMaxOrder()) {
+            return;
+        }
+        if (!needsToBeLogged(idealMaxOrder)) {
+            return;
+        }
+
+        if (idealMaxOrder > NETTY_MAXIMUM_ORDER) {
+            logger.error("Got a batch size of {} bytes that can fit into maximum maxOrder value 14, can't increment more", requiredSize);
+        } else {
+            logger.warn("Got a batch size of {} bytes, while this instance expects batches up to {}, please bump maxOrder to {}.",
+                    requiredSize, PooledByteBufAllocator.DEFAULT.metric().chunkSize(), idealMaxOrder);
```

Comment:
```suggestion
            logger.warn("Received batch of size {} bytes that is too large to fit into the pre-allocated memory pool. This will cause a performance degradation. Set 'io.netty.allocator.maxOrder' JVM property to {} to accommodate batches bigger than {} bytes.",
                    requiredSize, idealMaxOrder, PooledByteBufAllocator.DEFAULT.metric().chunkSize());
```